### ISSUE TITLE
error while opening EN0SPC

### ISSUE THREAD

**Issue by bastos77** [keeweb/keeweb#1439]
![Bildschirmfoto von 2020-03-31 13-05-06](https://user-images.githubusercontent.com/12494783/78019609-4c208900-7350-11ea-925e-3d107c517a50.png)

This error "System limit for number of file watchers reached" appears when I try to open my kdbx stored in a folder which is synchonized with OwnCloud.
But after clicking "OK" everything is usable.
Do I have to be concerned?

KeeWeb v1.12.3 (9b07bbd5, 2019-11-06)
Environment: electron v6.0.2
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) KeeWeb/1.9.0 Chrome/76.0.3809.110 Electron/6.0.2 Safari/537.36

**Comment by antelle** [comment]
Hi! This shouldn't happen indeed. Could you please post logs if this happens again?

**Comment by bastos77** [comment]
Strange. Happened the last weeks every time I opened my kdbx but today, first opening after posting this issue, everything went right...

**Comment by antelle** [comment]
ok, if this happens again, please add a comment here, better with logs. Closing it for now

**Comment by bastos77** [comment]
Happened again:
```
[INFO ] 2020-04-01T16:21:30.851Z [app] Started in 679ms ¯\_(ツ)_/¯
[INFO ] 2020-04-01T16:21:30.854Z [updater] Next update check will happen in 1s
[INFO ] 2020-04-01T16:21:31.355Z [updater] Checking for update...
[INFO ] 2020-04-01T16:21:31.356Z [transport] GET https://app.keeweb.info/manifest.appcache
[INFO ] 2020-04-01T16:21:31.377Z [transport] Request to https://app.keeweb.info/manifest.appcache without proxy
[INFO ] 2020-04-01T16:21:31.822Z [transport] Response from https://app.keeweb.info/manifest.appcache: 200
[INFO ] 2020-04-01T16:21:31.823Z [updater] Update check: # 2019-11-06:v1.12.3
[INFO ] 2020-04-01T16:21:31.824Z [updater] Next update check will happen in 86400s
[INFO ] 2020-04-01T16:21:41.308Z [open:[filename]] File open request
[INFO ] 2020-04-01T16:21:41.308Z [open:[filename]] Open file from storage file
[INFO ] 2020-04-01T16:21:41.308Z [open:[filename]] Stat file
[DEBUG] 2020-04-01T16:21:41.308Z [storage-file] Stat /[path]
[DEBUG] 2020-04-01T16:21:41.310Z [storage-file] Stat done /[path][filename] 2ms
[INFO ] 2020-04-01T16:21:41.311Z [open:[filename]] Open file from storage (1585653710572, local 1585653710572)
[INFO ] 2020-04-01T16:21:41.311Z [open:[filename]] Load from storage
[DEBUG] 2020-04-01T16:21:41.311Z [storage-file] Load /[path][filename]
[DEBUG] 2020-04-01T16:21:41.322Z [storage-file] Loaded /[path][filename] 1585653710572 11ms
[INFO ] 2020-04-01T16:21:41.323Z [open:Sebastian-key] Open file from content loaded from storage
[ERROR] 2020-04-01T16:21:41.398Z [file] Error opening file InvalidKey Error InvalidKey KdbxError: Error InvalidKey
[ERROR] 2020-04-01T16:21:41.399Z [open-view] Error opening file KdbxError: Error InvalidKey
[INFO ] 2020-04-01T16:21:53.753Z [open:[filename]] File open request
[INFO ] 2020-04-01T16:21:53.753Z [open:[filename]] Open file from supplied content
[INFO ] 2020-04-01T16:21:54.345Z [file] Opened file [filename]: 580ms, R=300000, 54 kB
[DEBUG] 2020-04-01T16:21:54.345Z [app] Add last open file 19f966fb-a400-3175-5abb-9f4d4d266eb5 [filename] file /[path][filename] 1585653710572
[DEBUG] 2020-04-01T16:21:54.579Z [storage-file] Watch dir [path]
[ERROR] 2020-04-01T16:21:54.581Z [file] Error opening file ENOSPC ENOSPC: System limit for number of file watchers reached, watch '[path]' Error: ENOSPC: System limit for number of file watchers reached, watch '[path]'
[ERROR] 2020-04-01T16:21:54.581Z [open-view] Error opening file Error: ENOSPC: System limit for number of file watchers reached, watch '[path]'
```

**Comment by antelle** [comment]
Hm, it should not complain in this case, could you please open devtools and copy the error stack (if you expand the error)? Devtools can be opened from settings / general / advanced.

**Comment by bastos77** [comment]
This one?
```
2020-04-01T16:38:43.446Z [file] Error opening file ENOSPC ENOSPC: System limit for number of file watchers reached, watch '[path]' Error: ENOSPC: System limit for number of file watchers reached, watch '[path]'
    at FSWatcher.start (internal/fs/watchers.js:165)
    at Object.watch (fs.js:1270)
    at Object.createFsWatcher (index.html:1)
    at r.watch (index.html:1)
    at P.fileOpened (index.html:1)
    at c (index.html:1)
    at index.html:1
a.error @ index.html:1
index.html:1 2020-04-01T16:38:43.446Z [open-view] Error opening file Error: ENOSPC: System limit for number of file watchers reached, watch '[path]'
    at FSWatcher.start (internal/fs/watchers.js:165)
    at Object.watch (fs.js:1270)
    at Object.createFsWatcher (index.html:1)
    at r.watch (index.html:1)
    at P.fileOpened (index.html:1)
    at c (index.html:1)
    at index.html:1
a.error @ index.html:1
manifest.json:1 GET file:///home/sebastian/.config/KeeWeb/app.asar/manifest.json net::ERR_FILE_NOT_FOUND
```

**Comment by antelle** [comment]
Perfect, thanks! This is already fixed in 1072e5778f4d4108cf44b417719dad44c1ca6a49 and coming in the next version.

**Comment by bastos77** [comment]
Thanks!

:-)

Am 01.04.20 um 18:47 schrieb Dimitri Witkowski:
